services/web/client/source/class/osparc/desktop/StudyEditor.js

```diff
@@ -135,6 +154,153 @@ qx.Class.define("osparc.desktop.StudyEditor", {
       this.__slideshowView.setStudy(study);
     },
 
+
+    // ------------------ START/STOP PIPELINE ------------------
+    __startPipeline: function(runAll = true) {
+      if (!osparc.data.Permissions.getInstance().canDo("study.start", true)) {
+        return;
+      }
+
+      const startStopButtonsWB = this.__workbenchView.getStartStopButtons();
+      const startStopButtonsSS = this.__slideshowView.getStartStopButtons();
+      startStopButtonsWB.setRunning(true);
+      startStopButtonsSS.setRunning(true);
+      this.updateStudyDocument(true)
+        .then(() => {
+          this.__doStartPipeline(runAll);
+        })
+        .catch(() => {
+          this.getLogger().error(null, "Run failed");
+          startStopButtonsWB.setRunning(false);
+          startStopButtonsSS.setRunning(false);
+        });
+    },
+
+    __doStartPipeline: function(runAll) {
+      if (this.getStudy().getSweeper().hasSecondaryStudies()) {
+        const secondaryStudyIds = this.getStudy().getSweeper().getSecondaryStudyIds();
+        secondaryStudyIds.forEach(secondaryStudyId => {
+          this.__requestStartPipeline(secondaryStudyId);
+        });
+      } else if (runAll) {
+        this.__requestStartPipeline(this.getStudy().getUuid());
+      } else {
+        const selectedNodeIDs = this.getPageContext() === "workbench" ? this.__workbenchView.getSelectedNodeIDs() : this.__slideshowView.getSelectedNodeIDs();
+        if (selectedNodeIDs === null || selectedNodeIDs.length === 0) {
+          this.__requestStartPipeline(this.getStudy().getUuid());
+        } else {
+          this.__requestStartPipeline(this.getStudy().getUuid(), selectedNodeIDs);
+        }
+      }
+    },
+
+    __requestStartPipeline: function(studyId, selectedNodeIDs = []) {
+      const url = "/computation/pipeline/" + encodeURIComponent(studyId) + ":start";
+      const req = new osparc.io.request.ApiRequest(url, "POST");
+      const startStopButtonsWB = this.__workbenchView.getStartStopButtons();
+      const startStopButtonsSS = this.__slideshowView.getStartStopButtons();
+      req.addListener("success", this.__onPipelinesubmitted, this);
+      req.addListener("error", e => {
+        this.getLogger().error(null, "Error submitting pipeline");
+        startStopButtonsWB.setRunning(false);
+        startStopButtonsSS.setRunning(false);
+      }, this);
+      req.addListener("fail", e => {
+        if (e.getTarget().getResponse().error.status == "403") {
+          this.getLogger().error(null, "Pipeline is already running");
+        } else {
+          this.getLogger().error(null, "Failed submitting pipeline");
+        }
+        startStopButtonsWB.setRunning(false);
+        startStopButtonsSS.setRunning(false);
+      }, this);
+      if (selectedNodeIDs.length) {
+        req.setRequestData({
+          "subgraph": selectedNodeIDs
+        });
```

Comment:
you can add the force_restart field here.. once https://github.com/ITISFoundation/osparc-simcore/pull/2094 goes in

Reply:
Exactly 💪